inferno-core/src/Inferno/Module/Prelude.hs

```diff
@@ -587,6 +589,16 @@ module Word
   @doc Convert a word to an `int`;
   fromWord : forall 'a. {requires fromWord on 'a} => 'a -> int := ###fromWordFun###;
 
+  define fromBCD on word64;
+
+  @doc Decode a BCD-encoded word64;
+  fromBCD : forall 'a. {requires fromBCD on 'a} => 'a -> option of word64 := ###fromBCDFun###;
```

Comment:
You don't need the type class `fromBCD` unless you're going to allow the function to be applied to more than the one type `word64`, right?